crates/polars-core/src/chunked_array/ops/sort/mod.rs

```diff
@@ -534,6 +541,19 @@ impl ChunkSort<BooleanType> for BooleanChunked {
     }
 }
 
+#[cfg(feature = "object")]
```

Comment:
This whole impl can be removed. You can set the `polars_bail(opq = sort)` on the `SeriesTrait`. That means you also don't have to put the `PolarsResult` the `ChunkedArrays` that do support it.

Reply:
So remove this impl, set the `polars_bail(opq = *)` on both the `arg_sort` and `sort_with` within the `impl<T> SeriesTrait for SeriesWrap<ObjectChunked<T>>`, then I can all the changes on the other types within `series/implementations/*`, right?

Reply:
Wait, on second guess I think with "you can set the polars_bail(opq = sort) on the SeriesTrait", you mean on `pub trait SeriesTrait`, but I have already done that in this PR. That leads to changing the return type to `PolarsResult `, which leads to all the changes in the other files like I did.

If you do mean what I initially thought (to put the `polars_bail` on the `impl<T> SeriesTrait for SeriesWrap<ObjectChunked<T>>`) functions `arg_sort` and `sort_with`, then we will have the issue that the functions return type is different than the one defines in the pub `pub trait SeriesTrait` @ritchie46

Reply:
Yes, but I want it only on `SeriesTrait`. The implementors of sort are infallible. So they can return `Self`. 

And then we simply remove the implementations of sort for the dtypes that we don't support.